masterbase/lib.py

```diff
@@ -35,9 +38,33 @@ def make_db_uri(is_async: bool = False) -> str:
     return f"{prefix}://{user}:{password}@{host}:{port}/demos"
 
 
-def make_demo_path(session_id: str) -> str:
-    """Make the demo path for the current session."""
-    return os.path.join(DEMOS_PATH, f"{session_id}.dem")
+class DemoSessionManager(BaseModel):
+    """Helper class to facilitate access."""
+
+    session_id: str
+    detection_state: DetectionState
+    handle: Any = Field(default=None)
```

Comment:
Not a big fan of the Any here. What makes it difficult to know the type of this field statically? Are we able to restrict the type scope in any way?

Reply:
yeah this kind of sucks, I dont think pydantic likes file handles as attributes because it cant really validate those -- might be better to just swap to native dataclasses here

Reply:
we will always be opening demos as bytes right?? `open('file.dem', 'rb')`? So the return can be a `typing` `BinaryIO`. Or no?

Reply:
I couldnt get pydantic to initialize with `BinaryIO` or `IO[bytes]` which led to the code we see today haha